### PR TITLE
Stream Testkit: new-API-friendly 

### PR DESCRIPTION
**Pull request by ennru**
Add alternatives that use `ClassicActorSystemProvider` to allow for the use of the new actors API when testing streams.

- enable Mima on akka-stream-testkit
- add `apply`/`create` methods to complement 
	- `TestSink.probe`
	- `TestSource.probe`
	- `TestPublisher.probe`
	- `TestPublisher.manualProbe`
	- `TestSubscriber.probe`
	- `TestSubscriber.manualProbe`

The third commit uses the new variants at some random places.
- [ ] should all use in Akka be switched?
- [ ] should the old variants become deprecated (I'm not in favour)

This is a new variant of the non-binary compatible #29815

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM (I'm, fine with not deprecating and not updating every single use)

**Comment by raboof** [comment]
>     * [ ]  should all use in Akka be switched?
>     * [ ]  should the old variants become deprecated (I'm not in favour)

Hmm, I think we should eventually converge on one consistent style to use and show in docs, but I'm OK with having both live side-by-side for now.

> This is a new variant of the non-binary compatible #29809

That's #29815 :)

Since apparently telemetry was instrumenting the old methods, should we notify them that they might also want to instrument the new API's?

**Review by patriknw — APPROVED** [review]
LGTM, but I think we should create an issue targeted for Akka 2.7.0 to deprecated the old variants.

**Comment by patriknw** [comment]
> Since apparently telemetry was instrumenting the old methods, should we notify them that they might also want to instrument the new API's?

^ this is important

**Comment by johanandren** [comment]
Was telemetry really instrumenting, wasn't it about using the testkit from one of our other akka-umbrella modules which telemetry pulls in and needing bincomp for that?

**Comment by pyoio** [comment]
Apologies for the delay in responding, had to dust some cobwebs off the jenkins job for this purpose, https://ci.lightbend.com/view/Cinnamon/job/cinnamon-akka-pr-validation/22/ is currently running (against this branch) and should indicate that the changes are ✅

**Comment by pyoio** [comment]
The PR job failed (for acceptable reasons) -- I think this PR doesn't contain the changes merged in https://github.com/akka/akka/pull/29823 so the failing cinnamon assertions are expected where actual vs. expected stream events differ.

TL;DR: good to go 👍  @ennru @patriknw
